### PR TITLE
Add transaction/retry support for yii\db\Command

### PR DESCRIPTION
**Pull request by sergeymakinen**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | see #15276 

It's required to perform upsert statements safely and securely.

This PR has to be merged before #15276.

### CODE REVIEW

File: framework/db/Command.php [hunk 1]
```diff
@@ -1114,7 +1124,78 @@ protected function refreshTableSchema()
...
+        $this->_isolationLevel = $isolationLevel;
+        return $this;
+    }
+
+    /**
+     * Sets a callable (e.g. anonymous function) that is called when [[Exception]] is thrown
+     * when executing the command. The signature of the callable should be:
+     *
+     * ```php
+     * function (\yii\db\Exception $e, $attempt)
```
Comment (samdark) [inline framework/db/Command.php]: Would be good to describe `$e` and `$attempt`.

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -50,6 +50,10 @@ Yii Framework 2 Change Log
 - Enh #15347: Add `Instance` support for object property in DI container (kojit2009)
 - Enh #15357: Added multi statement support for `yii\db\sqlite\Command` (sergeymakinen)
 - Enh #15360: Refactored `BaseConsole::updateProgress()` (developeruz)
+- Bug #15317: Regenerate CSRF token if an empty value is given (sammousa)
+- Bug #15380: `FormatConverter::convertDateIcuToPhp()` now converts `a` ICU symbols to `A` (brandonkelly)
```
Comment (samdark) [inline framework/CHANGELOG.md]: Changelog should be fixed.

Comment (sergeymakinen) [reply framework/CHANGELOG.md]: Oh, it always happens. :<(

File: framework/db/Command.php (hunk 1 above)
Comment (sergeymakinen) [reply framework/db/Command.php]: Done.

**Review by samdark — APPROVED** [review]

**Comment by schmunk42** [comment]
I haven't found the retry part in this PR (or I didn't get it). But I wanted to link to this [retry](https://github.com/igorw/retry) lib for PHP, which is really great - just in case it might be useful here.

**Comment by sergeymakinen** [comment]
@schmunk42 Well, [it](https://github.com/yiisoft/yii2/pull/15415/files#diff-3898926ea9b3c7307c9d94389ab91b7bR1177) works the same, you may want to start [here](https://github.com/yiisoft/yii2/pull/15415/files#diff-e48002914d2773120698a96387ecabd2R1163) for an usage example.

**Review by SilverFire — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**